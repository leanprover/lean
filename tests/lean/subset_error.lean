constant set    : Type₁ → Type₁
constant finset : Type₁ → Type₁
constant set.mem : Π {A : Type₁}, A → set A → Prop
constant finset.mem : Π {A : Type₁}, A → finset A → Prop

infix ∈ := set.mem
infix ∈ := finset.mem

definition set.subset {A : Type₁} (s₁ s₂ : set A) : Prop :=
∀ ⦃a : A⦄, a ∈ s₁ → a ∈ s₂

definition finset.subset {A : Type₁} (s₁ s₂ : finset A) : Prop :=
∀ ⦃a : A⦄, a ∈ s₁ → a ∈ s₂

infix `⊆` := set.subset
infix `⊆` := finset.subset

example (A : Type₁) (x : A) (S H : set A) (Pin : x ∈ S) (Psub : S ⊆ H) : x ∈ H :=
Psub Pin -- Error, we cannot infer at preprocessing time the binder information for Psub
