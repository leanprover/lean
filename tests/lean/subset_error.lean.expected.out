subset_error.lean:18:51: error: type mismatch at application
  x âˆˆ S
term
  S
has type
  set A
but is expected to have type
  finset A
subset_error.lean:18:51: previous error additional information
none of the overloads is applicable: finset.mem set.mem
subset_error.lean:18:66: previous error additional information
none of the overloads is applicable: finset.subset set.subset
